# 🛒 Superstore Sales — Exploratory Data Analysis (EDA)

**Goal:** Explore the Superstore Sales dataset to understand sales patterns across categories, regions, segments, and shipping modes — and extract actionable business insights.

**Dataset:** `train.csv` (Superstore Sales) — 9,800 rows × 18 columns

**Workflow:**
1. Load & Inspect Data
2. Check for Missing Values & Duplicates
3. Clean & Format Data
4. Statistical Summary
5. Business Analysis (Groupby Insights)
6. Key Findings & Conclusion

## 1️⃣ Load & Inspect the Data

Load the dataset and take a first look at its structure — shape, columns, and data types.

In [1]:
import pandas as pd

df = pd.read_csv('train.csv', encoding='latin1')  # encoding needed for this dataset

print(df.head())
print(df.shape)
print(df.info())

   Row ID        Order ID  Order Date   Ship Date       Ship Mode Customer ID  \
0       1  CA-2017-152156  08/11/2017  11/11/2017    Second Class    CG-12520   
1       2  CA-2017-152156  08/11/2017  11/11/2017    Second Class    CG-12520   
2       3  CA-2017-138688  12/06/2017  16/06/2017    Second Class    DV-13045   
3       4  US-2016-108966  11/10/2016  18/10/2016  Standard Class    SO-20335   
4       4  US-2016-108966  11/10/2016  18/10/2016  Standard Class    SO-20335   

     Customer Name    Segment        Country             City       State  \
0      Claire Gute   Consumer  United States        Henderson    Kentucky   
1      Claire Gute   Consumer  United States        Henderson    Kentucky   
2  Darrin Van Huff  Corporate  United States      Los Angeles  California   
3   Sean O'Donnell   Consumer  United States  Fort Lauderdale     Florida   
4   Sean O'Donnell   Consumer  United States  Fort Lauderdale     Florida   

   Postal Code Region       Product ID         Cat

> **Observation:** The dataset has 9,800 rows and 18 columns. Most columns are text (`object`), with only `Row ID`, `Postal Code`, and `Sales` as numeric. `Order Date` and `Ship Date` are stored as text and will need conversion to proper dates.

## 2️⃣ Check for Missing Values & Duplicates

Before cleaning, confirm which columns have missing data and whether any duplicate rows exist.

In [2]:
# Check missing values
print(df.isnull().sum())

# Check duplicates
print("Duplicates:", df.duplicated().sum())

# Look at unique values in key categorical columns
print(df['Category'].unique())
print(df['Segment'].unique())
print(df['Region'].unique())

Row ID            0
Order ID          0
Order Date        0
Ship Date         0
Ship Mode         0
Customer ID       0
Customer Name     0
Segment           0
Country           0
City              0
State             0
Postal Code      11
Region            0
Product ID        0
Category          0
Sub-Category      0
Product Name      0
Sales             0
dtype: int64
Duplicates: 0
['Furniture' 'Office Supplies' 'Technology']
['Consumer' 'Corporate' 'Home Office']
['South' 'West' 'Central' 'East']


> **Observation:** Only `Postal Code` has missing values (11 rows). There are **0 duplicate rows**. The dataset has 3 product categories, 3 customer segments, and 4 regions.

## 3️⃣ Data Cleaning

- Fill missing `Postal Code` values (not critical for this analysis)
- Convert `Order Date` and `Ship Date` to proper datetime format

In [3]:
# Fill missing postal codes with 0 (placeholder, since we won't analyze postal code deeply)
df['Postal Code'] = df['Postal Code'].fillna(0)

# Convert Order Date and Ship Date to proper datetime format
df['Order Date'] = pd.to_datetime(df['Order Date'], format='%d/%m/%Y')
df['Ship Date'] = pd.to_datetime(df['Ship Date'], format='%d/%m/%Y')

# Confirm the changes
print(df.isnull().sum())
print(df.dtypes)

Row ID           0
Order ID         0
Order Date       0
Ship Date        0
Ship Mode        0
Customer ID      0
Customer Name    0
Segment          0
Country          0
City             0
State            0
Postal Code      0
Region           0
Product ID       0
Category         0
Sub-Category     0
Product Name     0
Sales            0
dtype: int64
Row ID                    int64
Order ID                 object
Order Date       datetime64[ns]
Ship Date        datetime64[ns]
Ship Mode                object
Customer ID              object
Customer Name            object
Segment                  object
Country                  object
City                     object
State                    object
Postal Code             float64
Region                   object
Product ID               object
Category                 object
Sub-Category             object
Product Name             object
Sales                   float64
dtype: object


> **Result:** No missing values remain. `Order Date` and `Ship Date` are now proper `datetime64` columns, ready for time-based analysis.

## 4️⃣ Statistical Summary

Examine the distribution of `Sales` and the frequency of key categorical variables.

In [4]:
# Summary stats for numerical columns
print(df[['Sales']].describe())

# Distribution of categorical columns
print(df['Category'].value_counts())
print(df['Segment'].value_counts())
print(df['Region'].value_counts())
print(df['Ship Mode'].value_counts())

              Sales
count   9800.000000
mean     230.769059
std      626.651875
min        0.444000
25%       17.248000
50%       54.490000
75%      210.605000
max    22638.480000
Category
Office Supplies    5909
Furniture          2078
Technology         1813
Name: count, dtype: int64
Segment
Consumer       5101
Corporate      2953
Home Office    1746
Name: count, dtype: int64
Region
West       3140
East       2785
Central    2277
South      1598
Name: count, dtype: int64
Ship Mode
Standard Class    5859
Second Class      1902
First Class       1501
Same Day           538
Name: count, dtype: int64


> **Observation:** `Sales` is **right-skewed** — the mean (230.77) is much higher than the median (54.49), pulled up by a small number of very large orders (max: 22,638.48). Office Supplies has the most orders, Consumer is the largest segment, West is the largest region, and Standard Class is the most-used shipping method.

## 5️⃣ Business Analysis (Groupby Insights)

Answer key business questions using `groupby` aggregations.

In [5]:
# Q1: Total sales by category
print(df.groupby('Category')['Sales'].sum().sort_values(ascending=False))

# Q2: Average sales by segment
print(df.groupby('Segment')['Sales'].mean().sort_values(ascending=False))

# Q3: Total sales by region
print(df.groupby('Region')['Sales'].sum().sort_values(ascending=False))

# Q4: Top 5 states by total sales
print(df.groupby('State')['Sales'].sum().sort_values(ascending=False).head())

# Q5: Which ship mode generates the most total sales?
print(df.groupby('Ship Mode')['Sales'].sum().sort_values(ascending=False))

Category
Technology         827455.8730
Furniture          728658.5757
Office Supplies    705422.3340
Name: Sales, dtype: float64
Segment
Home Office    243.403309
Corporate      233.150720
Consumer       225.065777
Name: Sales, dtype: float64
Region
West       710219.6845
East       669518.7260
Central    492646.9132
South      389151.4590
Name: Sales, dtype: float64
State
California      446306.4635
New York        306361.1470
Texas           168572.5322
Washington      135206.8500
Pennsylvania    116276.6500
Name: Sales, dtype: float64
Ship Mode
Standard Class    1.340831e+06
Second Class      4.499142e+05
First Class       3.455723e+05
Same Day          1.252190e+05
Name: Sales, dtype: float64


> **Key Insight:** Office Supplies has the **most orders** (5,909), but **Technology generates the most revenue** (827,455.87). This shows that order volume and revenue don't always align — Technology items are fewer in number but higher in value per order.

> **Segment Insight:** Consumer has the most orders overall, but **Home Office** customers spend the most on average per order (243.40).

## 6️⃣ Key Findings & Conclusion

The Superstore dataset contains **9,800 records and 18 columns**, with **no duplicate rows** and only 11 missing values (in `Postal Code`, which is not critical for this analysis). Sales are **right-skewed**, with the mean (230.77) exceeding the median (54.49) due to a small number of high-value transactions. **Technology** is the highest revenue-generating category, while **Office Supplies** records the greatest number of orders — showing that order volume and revenue do not always move together. Among regions, the **West** performs best in total sales, and **California** leads all states in revenue. The **Home Office** segment has the highest average order value, while **Consumer** contributes the most orders overall. **Standard Class** is the dominant shipping method by both order count and total sales.

**Business takeaway:** Prioritizing strategy purely by order volume can be misleading — high-frequency categories like Office Supplies don't necessarily drive the most revenue. Understanding both volume and value is essential for accurate business decision-making.